diff -rupN a/tools/build/src/engine/execcmd.c b/tools/build/src/engine/execcmd.c
--- a/tools/build/src/engine/execcmd.c	2015-08-11 06:53:51.000000000 -0700
+++ b/tools/build/src/engine/execcmd.c	2020-12-09 09:39:49.000000000 -0800
@@ -10,6 +10,7 @@
 
 #include "jam.h"
 #include "execcmd.h"
+#include "output.h"
 
 #include <assert.h>
 #include <stdio.h>
diff -rupN a/tools/build/src/engine/make.c b/tools/build/src/engine/make.c
--- a/tools/build/src/engine/make.c	2015-08-11 06:53:51.000000000 -0700
+++ b/tools/build/src/engine/make.c	2020-12-09 09:39:03.000000000 -0800
@@ -39,6 +39,7 @@
 #include "headers.h"
 #include "lists.h"
 #include "object.h"
+#include "output.h"
 #include "parse.h"
 #include "rules.h"
 #include "search.h"
diff -rupN a/tools/build/src/engine/modules/path.c b/tools/build/src/engine/modules/path.c
--- a/tools/build/src/engine/modules/path.c	2015-08-11 06:53:51.000000000 -0700
+++ b/tools/build/src/engine/modules/path.c	2020-12-09 09:42:12.000000000 -0800
@@ -5,6 +5,7 @@
  */
 
 #include "../constants.h"
+#include "../filesys.h"
 #include "../frames.h"
 #include "../lists.h"
 #include "../native.h"
diff -rupN b/tools/build/src/engine/filesys.h c/tools/build/src/engine/filesys.h
--- b/tools/build/src/engine/filesys.h	2021-01-19 13:25:37.000000000 -0800
+++ c/tools/build/src/engine/filesys.h	2021-01-19 13:25:51.000000000 -0800
@@ -104,4 +104,7 @@ void file_query_posix_( file_info_t * co
 
 void file_done();
 
+int filelist_empty( FILELIST * list );
+int file_collect_archive_content_( file_archive_info_t * const archive );
+
 #endif
